### ISSUE TITLE
jspm install npm:alt fails - trouble with main field pointing to directory

### ISSUE THREAD

**Issue by arackaf** [jspm/jspm-cli#1114]
jspm install npm:alt successfully installs alt, but when attempting to use it, I'm getting a 404 on a lib file.  Looking in alt's package.json reveals this

  "main": "lib",

and of course inside of the lib directory is an index.js file.  It appears as though jspm isn't written to cover this use case?

Apologies if this is a duplicate; I read through the list of issues and didn't see anything like this.

**Comment by arackaf** [comment]
So sorry - I had been playing around with this quite a bit, and the steps described above weren't quite right.

The default npm install for alt has a main field in package.json which points to lib.  Inside of the lib directory is an index.js file.

The jspm install of alt has a lib.js file with this as its content:

module.exports = require('./lib/index');

When I try to load this module up, the lib.js file does come down correctly, but then I'm seeing a 403 forbidden error on the lib directory.  I've reproduced this on both Chrome and Firefox on Windows 10.

**Comment by guybedford** [comment]
@arackaf do you have `defaultJSExtensions: true` set here?

**Comment by arackaf** [comment]
@guybedford - yes, it appears so.  This is what config.js looks like.

Is something else malformed? 

System.config({
  baseURL: "/",
  defaultJSExtensions: true,
  transpiler: "none",
  paths: {
    "npm:*": "jspm_packages/npm/*"
  },

  map: {
    "alt": "npm:alt@0.17.3",
    "npm:alt@0.17.3": {
      "flux": "npm:flux@2.0.3",
      "transmitter": "npm:transmitter@1.0.2"
    }
  }
});

**Comment by guybedford** [comment]
It sounds like something else must be requiring `alt/lib` directly by name, so solving this would be to track that down. I think the main is working fine though, so it must be some relative require internall like `require('../lib')` or a parent dependency trying to load `require('alt/lib')`. If we could find what file is doing this, we can then use a map config override to correct the issue.

**Comment by arackaf** [comment]
@guybedford - thanks for the reply.  Really the point of this thread was to report what I thought was a bug in jspm.  If I'm wrong, and Alt is doing something wrong, then by all means close this, but I do suspect it's the former.

> It sounds like something else must be requiring alt/lib

Indeed.  It's the very entry point into Alt that jspm created:

    module.exports = require("npm:alt@0.17.3/lib");

Which is in alt@0.17.3.js

Some further info: Alt's package.json's main value points to `lib` and inside of the lib directory is an index.js.  At the risk of stating the obvious I suspect something inside of index.js is triggering the `require("npm:alt@0.17.3/lib");` which is failing for some reason.

**Comment by telekosmos** [comment]
FYI, I didn't use 0.17.3, but I'm using alt@0.17
```javascript
"dependencies": {
      "alt": "npm:alt@^0.17"
      //, ...
}
```
and it is working ok. Maybe there is a bug in 0.17.3 build in order to use it with jspm...
Just in case you want to downgrade; maybe you should report the `alt.js` team about this issue....

**Comment by guybedford** [comment]
I think the index is working fine (we create `lib.js` for this reason), the issue specifically would apply to a require of `require('alt/lib/')` with a trailing slash if you can find what is doing that. Note that this will be fixed in jspm 0.17 as we are improving the Node conversion process. If you want further help tracking it down I'm happy to assist with the override, but apart from that it will be about testing this out again with jspm 0.17 when that is released.

**Comment by arackaf** [comment]
@guybedford - sounds great.  I'll give this another test when jspm 0.17 is out.

Is there an estimate on when that'll be?
